http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -1261,7 +1262,41 @@ class LeoPubsubMessageSubscriber[F[_]](
                   }
               else F.unit
 
-            _ <- List(deleteDataDisk, deletePostgresDisk).parSequence_
+            deleteStagingBucket = dbApp.app.appType match {
+              case AppType.Allowed =>
```

Comment:
hmm... would it be possible to detect whether a staging bucket exists via `getGoogleStorageServiceFromRegistry()` rather than decide based on the appType here?

Reply:
we can just fire a delete, and ignore if it's 404..I don't think doing a GET first is necessary since it'll adds one more GCP call for not much benefit.
Calling based on app type is a bit of optimization because we know welder is only used for GCP apps today. But I'm happy to remove this logic if you think we'll likely support welder for other apps

Reply:
Yeah, if firing a delete if none is there doesn't have a downside as you say, I'd rather not optimize this logic and always try to clean it up to prevent us from accidentally leaking in the future